### ISSUE TITLE
Rehydration failure of serialized container with reference to offline blob

### ISSUE THREAD

**Issue by coclauso** [microsoft/FluidFramework#6828]
## Rehydration failure of serialized container with reference to offline blob
This bug concerns blobs for detached containers via `IDetachedBlobStorage`, discussed here #6128 implemented with #6530.

The observed behavior is that if a container referencing an offline blob is serialized (i.e. `Container.serialize()`) and then rehydrated (i.e., `Loader.rehydrateDetachedContainerFromSnapshot()`), the rehydration will fail, at this location/stack trace:
![image](https://user-images.githubusercontent.com/48933045/126430740-f3f282e7-e01f-4b06-893f-13b5c8b3492a.png)

Inspection of the serialized JSON allows identification of the relevant part:
```
               }
            },
            ".metadata":{
               "type":2,
               "content":"{\"summaryFormatVersion\":1,\"gcFeature\":0}"
            },
            ".blobs":{
               "type":1,
               "tree":{
                  "0":{
                     "id":"801a72e9-2251-42f4-bdb3-eafcda34fb32",
                     "type":4
                  }
               }
            }
         }
      }
   }
}
```

Full text not posted here.  More information can be provided over DM if helpful.

### To Reproduce
From a high level:
- Instantiate loader with IDetachedBlobStorage
- Create detached container from loader
- While container is detached, upload blob via runtime
- Serialize detached container, and rehydrate

### Expected behavior
Rehydration succeeds

Additional comment: to an implementer of `IDetachedBlobStorage`, it's not documented what should be returned for `url`.  Tried a few things in an attempt to rule this out as a factor, the above failure reproduced regardless.

**Comment by DLehenbauer** [comment]
convertSummaryToSnapshotWithEmbeddedBlobContents() isn't expecting SummaryType.Attachment.

**Comment by wes-carlson** [comment]
fixed in #6864
